include/llvm/MC/MCStreamer.h

```diff
@@ -458,6 +458,12 @@ class MCStreamer {
   /// \param Symbol - Symbol the section relative relocation should point to.
   virtual void EmitCOFFSecRel32(MCSymbol const *Symbol);
 
+  /// \brief Emits a COFF section relative relocation with an offset.
+  ///
+  /// \param Symbol - Symbol the section relative relocation should point to.
+  /// \param Offset - The offset from Symbol.
+  virtual void EmitCOFFSecRelOffset32(MCSymbol const *Symbol, int Offset);
```

Comment:
Can we define EmitCOFFSecRel32(MCExpr*) or EmitCOFFSecRel32Value(MCExpr*) so that we allow a general symbol expression, not just Offset addition? Once we have that, the common code of EmitCOFFSecRel32(MCSymbol*) should be factored out instead of duplicating the most of code.